llvm/lib/TargetParser/RISCVISAInfo.cpp

```diff
@@ -799,6 +800,14 @@ Error RISCVISAInfo::checkDependency() {
         errc::invalid_argument,
         "'zabha' requires 'a' or 'zaamo' extension to also be specified");
 
+  if (HasD && Exts.count("xwchc") != 0)
```

Comment:
IIRC `xwchc` also requires `XLen == 32`.

```suggestion
  if (XLen != 32 && Exts.count("xwchc"))
    return createStringError(errc::invalid_argument,
                             "'xwchc' is only supported for 'rv32'");

  if (HasD && Exts.count("xwchc") != 0)
```

Reply:
I don't believe the opcode encodings *conflict* with XLen != 32, but, as no 64-bit microcontrollers from this vendor currently exist, I will add this check.